js/try.js

```diff
@@ -94,6 +97,58 @@ const setCache = ({
   }
 }
 
+const doDownload = async (downloadButton) => {
+  try {
+    const {
+      identifier,
+    } = getCache();
+
+    let classes = [];
+    for (let i = 0; i <= 35; i++) { // say there's 35 weeks
+      const url = buildWeekURL(identifier, i);
+      downloadButton.textContent = `Fetching week ${i}`;
+      const timetable = await fetchAndParseTimetable(url);
```

Comment:
This will fetch the timetable 35 times, when it only needs to do it once, as it seems to retrieve the same data

Reply:
there are 2 approaches for fetching the data
the one you're using is fetching all the classes from the semester timetable, and then trying to figure out which academic weeks correspond to which days (which is tricky, because there are holidays and things like that...) by hard-coding the holiday lengths:

```
    const days = info[9].text
      .split(', ')
      .flatMap((range) => {
        if (range.indexOf('-') !== -1) {
          const [start, end] = range.split('-').map(Number)
          return Array(end - start + 1)
            .fill()
            .map((_, i) => i + start)
        }
        return Number(range)
      })
      .map((week) => {
        if (week <= 11) {
          return week
        } else if (week <= 22) {
          return week + 4
        }
        return week + 8
      })
```
this will break if the holidays change. i dont even know if it works for other (non-COMP) students...

so instead i fetch all the weeks separately, and parse the "official" date from the top left. this is less likely to go wrong if the holidays change, because the dates will always be accurate on the Leeds website

i realize making 35 requests is not ideal, but i guess people will just have to wait for a couple of seconds once a year

Reply:
These are consistent across the uni, it is from their [official website](https://ses.leeds.ac.uk/info/21630/timetabling/1004/teaching_week_patterns).

Reply:
If it needs to be changed in future years, that's not too much of an issue.